False positive: arguments-differ for __init_subclass__ in generic

### Bug description

```python
from typing import Generic, TypeVar


T = TypeVar("T")


class Testing(Generic[T]):  # pylint: disable=too-few-public-methods
    """Placeholder."""

    def __init_subclass__(cls, /, **kwargs: Any) -> None:
        """Placeholder."""
        super().__init_subclass__(**kwargs)
```

### Configuration

_No response_

### Command used

```shell
pylint file.py
```


### Pylint output

```shell
************* Module file
file.py:10:4: W0221: Variadics removed in overriding 'Testing.__init_subclass__' method (arguments-differ)

------------------------------------------------------------------
Your code has been rated at 9.88/10 (previous run: 9.88/10, +0.00)
```


### Expected behavior

I believe this is a false positive.

When I don't inherit from `Generic[T]`, the error goes away.

### Pylint version

```shell
pylint 2.17.5
astroid 2.15.6
Python 3.11.1 (v3.11.1:a7a450f84a, Dec  6 2022, 15:24:06) [Clang 13.0.0 (clang-1300.0.29.30)]
```


### OS / Environment

MacOS Ventura 13.5

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/8919.false_positive
+++ b/doc/whatsnew/fragments/8919.false_positive
@@ -1,0 +1,3 @@
+Fix false positive for ``arguments-differ`` when overriding `__init_subclass__`.
+
+Closes #8919

--- a/pylint/checkers/classes/class_checker.py
+++ b/pylint/checkers/classes/class_checker.py
@@ -370,14 +370,6 @@ def _different_parameters(
         if different_kwonly:
             output_messages += different_kwonly
 
-    if original.name in PYMETHODS:
-        # Ignore the difference for special methods. If the parameter
-        # numbers are different, then that is going to be caught by
-        # unexpected-special-method-signature.
-        # If the names are different, it doesn't matter, since they can't
-        # be used as keyword arguments anyway.
-        output_messages.clear()
-
     # Arguments will only violate LSP if there are variadics in the original
     # that are then removed from the overridden
     kwarg_lost = original.args.kwarg and not overridden.args.kwarg
@@ -385,6 +377,14 @@ def _different_parameters(
 
     if kwarg_lost or vararg_lost:
         output_messages += ["Variadics removed in"]
+
+    if original.name in PYMETHODS:
+        # Ignore the difference for special methods. If the parameter
+        # numbers are different, then that is going to be caught by
+        # unexpected-special-method-signature.
+        # If the names are different, it doesn't matter, since they can't
+        # be used as keyword arguments anyway.
+        output_messages.clear()
 
     return output_messages
 

--- a/tests/functional/a/arguments_differ.py
+++ b/tests/functional/a/arguments_differ.py
@@ -358,3 +358,13 @@ class AClass:
 class ClassWithNewNonDefaultKeywordOnly(AClass):
     def method(self, *, arg2, arg1=None):  # [arguments-differ]
         ...
+
+
+# Exclude `__init_subclass__` from the check:
+class InitSubclassParent:
+    def __init_subclass__(cls, *args, **kwargs):
+        ...
+
+class InitSubclassChild(InitSubclassParent):
+    def __init_subclass__(cls, /, **kwargs) -> None:
+        super().__init_subclass__(**kwargs)

